### PR TITLE
integration: use unix sockets for all connections

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: pkg/transport/transport.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import (
+	"net"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type unixTransport struct{ *http.Transport }
+
+func NewTransport(info TLSInfo, dialtimeoutd time.Duration) (*http.Transport, error) {
```
Comment (xiang90) [inline pkg/transport/transport.go]: NewUnixTransport?

Comment (heyitsanthony) [inline pkg/transport/transport.go]: but it does http too

**Comment by xiang90** [comment]
lgtm

**Comment by gyuho** [comment]
lgtm.

File: pkg/transport/transport.go (hunk 1 above)
Comment (xiang90) [inline pkg/transport/transport.go]: line 48

```go
return dialer.Dial("unix", addr)
```

doesnt the dialer always try unix rather than tcp?

Comment (heyitsanthony) [inline pkg/transport/transport.go]: sort of-- when the returned Transport sees the URL scheme is {unix,unixs}:// it uses a registered alternative roundtripper which changes the scheme "unix" with "http" and then calls to a second Transport with a unix dialer; if the scheme is http/https it calls the usual tcp net.Dialer{...}.Dial

Comment (xiang90) [inline pkg/transport/transport.go]: ah... LGTM... I was not aware of the RegisterProtocol thing.

**Pull request merged**